pkg/kubelet/apis/deviceplugin/v1beta1/api.proto

```diff
@@ -73,10 +73,16 @@ message ListAndWatchResponse {
 	repeated Device devices = 1;
 }
 
+message TopologyInfo {
+    	int64 socket = 1;
```

Comment:
Colleagues with more experience with gRPC APIs, correct me if I'm wrong here.

This change introduces new struct, which has `int64` field. Default value for this field if omitted, is `0` which is already potentially overlaps with valid value range for this field, but that's not critical at the moment.

Currently, if device should not be aligned with socket, plugin should not be setting this struct at all, meaning that `topology` will be `nil` and code in Topology Manager will use absence of this struct as a flag for non-aligned to the socket devices.

Now, imagine that in future we will add more topology info fields into that struct (NVlink or any other similar kind of "out-of-band" links). 
If plugin will be interested to use those new fields, but not `socket` field, `socket` will be uninitialised, which will be defaulted to `0` and lead to incorrect behaviour (device will be assumed as aligned against socket 0, even were device is socket agnostic). Thus to indicate non-socket aligned devices we will need to have some pre-agreed value (e.g. -1?), and plugins will be forced to explicitly set `socket` field to this pre-agreed value, even if they are not interested in it. Which is in turn means: not very friendly for consumers of this API (plugins).

Would it make more sense to have `socket` field as `string` from the beginning? Doing `ParseInt` on receiving side is easy, and if field is not explicitly set by the plugin, uninitialised field will be defaulted to empty string, which is a lot easier to use as an indicator for non-aligned to the sockets devices.

Reply:
This is a good point @kad. While a string could work, there's little precedent for it in the existing interfaces. Perhaps a nested struct (which would be nullable) could serve the same purpose to fix the problem you identified.

Reply:
@ConnorDoyle can you elaborate about "little precedent" part?

As for nested structs - it can work, but having nested struct for just one integer field is kind of overkill IMHO, especially taking into account that de-referencing of potential `nil` will require additional handling for each of such structs (parent TopologyInfo can be `nil` and any other sub-struct with meaningful fields can be `nil`).

Anyway, let's hear other comments. Maybe there are some other ways to it more future proof.

Reply:
It should be noted that for hardware where the NUMA affinity is not known, /sys/bus/pci/devices/<address>/numa_node will show as "-1" from the kernel.  So it might make sense to use -1 as a magic number for non-aligned devices.

Reply:
Do we have consensus on the format this should take? I agree a -1 would be preferable for the  device plugins that don't know or don't care about NUMA. However @kad is correct in that the default will be 0, and I am not aware of a way of setting a default value in proto.

So, should we go for a string value or a nested struct?

Reply:
I am in favor of wrapping the socket field in a `Socket` message like so:
```
message Socket {
    	int64 id = 1;
}
message TopologyInfo {
    	Socket socket = 1;
}
```

Sticking with the theme of making things future proof, this solves the immediate problem, as well as gives us the ability to extend the `Socket` message in the future if we want to hang any additional information off of it other than its ID.